### PR TITLE
Controller specs

### PR DESCRIPTION
**Pull request by Dadadon**
**
> Added RSpec as a gem to the project.
**
Created a Request spec file for controllers.
**
> For each action I checked:
**

- If the response status was correct.
- If a correct template was rendered.
- If the response body includes the correct placeholder text.

### CODE REVIEW

File: spec/requests/controllers_spec.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    it 'renders the show template' do
+      expect(response).to render_template(:show)
+    end
+
+    it 'If the response body includes correct placeholder text' do
+      expect(response.body).to include('Displays user info')
+    end
+  end
+end
```
Comment (CollinsTatang) [inline spec/requests/controllers_spec.rb]: Great job so far @Dadadon 

- [ ] Please it's required for you to have a separate `users_spec.rb` file to carry out tests for `index` and `show` actions.

File: spec/requests/controllers_spec.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      get "/users/#{user.id}/posts/#{post.id}"
+      expect(response).to render_template(:show)
+    end
+
+    it 'includes the post title and text in the response body' do
+      get "/users/#{user.id}/posts/#{post.id}"
+      expect(response.body).to include(post.title)
+    end
+  end
+end
```
Comment (CollinsTatang) [inline spec/requests/controllers_spec.rb]: Great job so far @Dadadon 

- [ ] Please it's required for you to have a separate `posts_spec.rb` file to carry out tests for `index` and `show` actions.

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @Dadadon, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @Dadadon 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**